sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/ConversationAsyncClient.java

```diff
@@ -469,17 +486,199 @@
+     * @return A {@link Mono} object containing the byte stream of the content requested.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Flux<ByteBuffer>> downloadStreaming(URI endpoint) {
+        return downloadStreamingWithResponse(endpoint, null).map(Response::getValue);
+    }
+
+    /**
+     * Download the recording content, e.g. Recording's metadata, Recording video, from the ACS endpoint
+     * passed as parameter.
+     * @param endpoint - URL where the content is located.
+     * @param httpRange - An optional {@link HttpRange} value containing the range of bytes to download. If missing,
+     *                  the whole content will be downloaded.
+     * @return A {@link Mono} object containing the byte stream of the content requested.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Flux<ByteBuffer>> downloadStreaming(URI endpoint, HttpRange httpRange) {
+        return downloadStreamingWithResponse(endpoint, httpRange).map(Response::getValue);
+    }
+
+    /**
+     * Download the recording content, (e.g. Recording's metadata, Recording video, etc.) from the {@code endpoint}.
+     * @param endpoint - URL where the content is located.
+     * @param range - An optional {@link HttpRange} value containing the range of bytes to download. If missing,
+     *                  the whole content will be downloaded.
+     * @return A {@link Mono} object containing a {@link Response} with the byte stream of the content requested.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Flux<ByteBuffer>>> downloadStreamingWithResponse(URI endpoint, HttpRange range) {
+        return downloadStreamingWithResponse(endpoint, range, null);
+    }
+
+    Mono<Response<Flux<ByteBuffer>>> downloadStreamingWithResponse(URI endpoint, HttpRange range, Context context) {
+        try {
+            Objects.requireNonNull(endpoint, "'endpoint' cannot be null");
+            return contentDownloader.downloadStreamingWithResponse(endpoint, range, context);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Download the recording content (e.g. Recording's metadata, Recording video, etc.) from the {@code endpoint}
+     * and written into the {@code stream}.
+     * @param stream - Stream to write the content to.
+     * @param endpoint - Location of the recording's content.
+     * @param range - An optional {@link HttpRange} value containing the range of bytes to download. If missing,
+     *                  the whole content will be downloaded.
+     * @param parallelDownloadOptions - An optional {@link ParallelDownloadOptions} object to modify how the
+     *                               parallel download will work.
+     * @return An empty {@link Mono} object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> downloadTo(
```

Comment:
Downloading to an OutputStream doesn't seem to fit with reactive programming, I'd recommend hiding/removing this API until there are requests for it. Especially given that the implementation uses a `.block()` internally which is generally a no-go for reactive APIs (may result in a runtime exception being thrown).

Reply:
[Before](https://apiview.dev/Assemblies/Conversation/73dbd47a7aa342fb8edaff8c38c0db74) you mentioned that I should return an [OutputStream], I didn't find a way to return the abstract class (probably I could return a `ByteArrayOutputStream` instead?) and given that I'm doing a similar method for the C# SDK, I thought to add it in this way.
If I'm given a similar API in C#, shouldn't I provide parallel functionality in Java as well?

Reply:
I hid this method to be used only by the sync client's `downloadTo` method.